client-protocol.md

```diff
@@ -103,3 +103,77 @@ will be released, and the WebSocket connection will be dropped.
 
 Now the client connection is fully set up, and the application specific messages
 (those with numeric phases) may be exchanged.
+
+## Wormhole Seeds
+
+Once two clients ever connected to each other, they now have a shared secret.
+This can be used to establish a new Wormhole connection without involving human
+entering codes. If A says "I want to connect to B" and B does the same they'll
+find each other and get a secure connection. Some additional data needs to be
+exchanged and stored in order to allow for a good user experience.
+
+Support for session resumption is declared using the
+`seeds-v1` ability during the `versions` phase. Additionally, a `seeds`
+key must be added to the versions message that roughly looks like this:
+
+```json
+{
+  "abilities": [ "seeds-v1" ],
+  "app_versions": {},
+  "seeds": {
+    "display_names": [<string>],
+    "known_seeds": [<string>],
+  },
+}
+```
+
+A client may choose a list of `display_names` in order to be recognizable. Note
+that client names may be arbitrary, collide with other sessions or change over
+time. Any valid UTF-8 string may be used as name, except for the following
+characters: `'`, `"` and `,`.
+
+It is up to the clients to keep track of such a mapping and keep it up to
+date, if they want to. It is also up to the clients to name themselves.
+We recommend giving at least two values: one with the user's
+name and one that also disambiguate multiple devices the user may have (now or in
+the future). The list must be sorted in decreasing order of preference.
+
+A seed is derived from the shared session key like this:
+
+```python
+# `derive(key, purpose)` is the usual key derivation function
+seed = hex(derive(session_key, "wormhole:seed"))
+```
+
+The `seed` is the main shared secret between the peers and all other data will
+be derived from it:
+
+```python
+password = hex(derive(seed, "wormhole:seed:password"))
+nameplate = hex(derive(seed, "wormhole:seed:nameplate"))
```

Comment:
Using a high entropy nameplate instead of connecting to a mailbox directly solves a lot of the connection issues that I previously had. Notably, multiple concurrent independent connections are possible using the same seed because they will end up in a different mailbox. Also, for the same reason, the error recovery when a mailbox gets "crowded" is a lot quicker.

The current rendezvous server implementation supports this without issues, but nevertheless it may be a good idea to codify this in the protocol that we now rely on this possibility.